### ISSUE TITLE
[wolfmax4k] (configuring) The SSL connection could not be established, see inner exception.: Unable to read data from the transport connection: Connection reset by peer.

### ISSUE THREAD

**Issue by Austros** [Jackett/Jackett#15520]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
Using HTTP Client: HttpWebC1ient/HttpWebC1ient2
Using FlareSoIverr: Whith url add in he setings (View in captures)
Using proxy: No
Running in Docker: Yes with image linuxserver/jackett:latest
Jackett variant: 64bit
Environment version: v0.22.409
```

### Description

Hi, I just installed the Jackett container along with Sonarr, Jellyfin, and Transmission on CasaOS, and when adding some indexers I get the following error: "An error occurred while configuring this indexer The SSL connection could not be established, see inner exception.: Unable to read data from the transport connection: Connection reset by peer." I've read some other issues and tried installing FlareSolverr but it didn't work. I've also read about updating OpenSSL, but the container that CasaOS installs doesn't have apt and I haven't been able to install it. Does anyone have any ideas? P.S. The indexer is Wolfmax 4k. Thanks in advance.

### Logged Error Messages

```text
jackett  | [migrations] started
jackett  | [migrations] no migrations found
jackett  | usermod: no changes
jackett  | ───────────────────────────────────────
jackett  | 
jackett  |       ██╗     ███████╗██╗ ██████╗
jackett  |       ██║     ██╔════╝██║██╔═══██╗
jackett  |       ██║     ███████╗██║██║   ██║
jackett  |       ██║     ╚════██║██║██║   ██║
jackett  |       ███████╗███████║██║╚██████╔╝
jackett  |       ╚══════╝╚══════╝╚═╝ ╚═════╝
jackett  | 
jackett  |    Brought to you by linuxserver.io
jackett  | ───────────────────────────────────────
jackett  | 
jackett  | To support LSIO projects visit:
jackett  | https://www.linuxserver.io/donate/
jackett  | 
jackett  | ───────────────────────────────────────
jackett  | GID/UID
jackett  | ───────────────────────────────────────
jackett  | 
jackett  | User UID:    1000
jackett  | User GID:    1000
jackett  | ───────────────────────────────────────
jackett  | Linuxserver.io version: v0.22.409-ls485
jackett  | Build-date: 2024-08-04T06:43:52+00:00
jackett  | ───────────────────────────────────────
jackett  |     
jackett  | [custom-init] No custom files found, skipping...
jackett  | 08-05 00:34:00 Info Starting Jackett v0.22.409
jackett  | 08-05 00:34:00 Info Environment version: 8.0.7 (/app/Jackett/)
jackett  | 08-05 00:34:00 Info OS version: Unix 6.8.0.39 (64bit OS) (64bit process)
jackett  | 08-05 00:34:00 Info Jackett variant: CoreLinuxMuslAmdx64
jackett  | 08-05 00:34:00 Info File /etc/issue: Welcome to Alpine Linux 3.20
jackett  | 08-05 00:34:00 Info Running in Docker: Yes (image build: v0.22.409-ls485)
jackett  | 08-05 00:34:00 Info ThreadPool MaxThreads: 32767 workerThreads, 1000 completionPortThreads
jackett  | 08-05 00:34:00 Info App config/log directory: /config/Jackett
jackett  | 08-05 00:34:00 Info Using proxy: Disabled
jackett  | 08-05 00:34:00 Info Using FlareSolverr: http://192.168.1.52:8191
jackett  | 08-05 00:34:00 Info Using HTTP Client: HttpWebClient2
jackett  | 08-05 00:34:00 Info Loading Native indexers ...
jackett  | 08-05 00:34:00 Info Loaded 72 Native indexers.
jackett  | 08-05 00:34:00 Info Loading Cardigann indexers from: /config/cardigann/definitions/, /etc/xdg/cardigan/definitions/, /app/Jackett/Definitions
jackett  | 08-05 00:34:02 Info Loaded 539 Cardigann indexers.
jackett  | 08-05 00:34:02 Info Loaded 611 indexers in total
jackett  | 08-05 00:34:02 Info Adding aggregate indexer ('all' indexer) ...
jackett  | 08-05 00:34:02 Info Adding filter indexer ('type:public' indexer) ...
jackett  | 08-05 00:34:02 Info Adding filter indexer ('type:private' indexer) ...
jackett  | 08-05 00:34:02 Info Adding filter indexer ('type:semi-public' indexer) ...
jackett  | 08-05 00:34:02 Info Jackett startup finished in 2.435 s
jackett  | Hosting environment: Production
jackett  | Content root path: /app/Jackett/Content
jackett  | Now listening on: http://[::]:9117
jackett  | Application started. Press Ctrl+C to shut down.
jackett  | Connection to localhost (::1) 9117 port [tcp/*] succeeded!
jackett  | [ls.io-init] done.
jackett  | 08-05 00:34:30 Warn Request to Wolfmax 4k failed with exception 'The SSL connection could not be established, see inner exception.'. Retrying in 2.6231892s.
jackett  | 08-05 00:34:32 Warn Request to Wolfmax 4k failed with exception 'The SSL connection could not be established, see inner exception.'. Retrying in 0.8133303s.
jackett  | 08-05 00:34:33 Error System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception.
jackett  | 
jackett  | [v0.22.409.0] System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception.
jackett  |  ---> System.IO.IOException: Unable to read data from the transport connection: Connection reset by peer.
jackett  |  ---> System.Net.Sockets.SocketException (104): Connection reset by peer
jackett  |    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.CreateException(SocketError error, Boolean forAsyncThrow)
jackett  |    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ReceiveAsync(Socket socket, CancellationToken cancellationToken)
jackett  |    at System.Net.Sockets.Socket.ReceiveAsync(Memory`1 buffer, SocketFlags socketFlags, Boolean fromNetworkStream, CancellationToken cancellationToken)
jackett  |    at System.Net.Sockets.NetworkStream.ReadAsync(Memory`1 buffer, CancellationToken cancellationToken)
jackett  |    at System.Net.Security.SslStream.EnsureFullTlsFrameAsync[TIOAdapter](CancellationToken cancellationToken, Int32 estimatedSize)
jackett  |    at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.ExecutionContextCallback(Object s)
jackett  |    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
jackett  |    at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.MoveNext()
jackett  |    at System.Net.Sockets.SocketAsyncEventArgs.TransferCompletionCallbackCore(Int32 bytesTransferred, Memory`1 socketAddress, SocketFlags receivedFlags, SocketError socketError)
jackett  |    at System.Net.Sockets.SocketAsyncEngine.System.Threading.IThreadPoolWorkItem.Execute()
jackett  |    at System.Threading.ThreadPoolWorkQueue.Dispatch()
jackett  |    at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
jackett  | --- End of stack trace from previous location ---
jackett  | 
jackett  |    --- End of inner exception stack trace ---
jackett  |    at System.Net.Security.SslStream.EnsureFullTlsFrameAsync[TIOAdapter](CancellationToken cancellationToken, Int32 estimatedSize)
jackett  |    at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
jackett  |    at System.Net.Security.SslStream.ReceiveHandshakeFrameAsync[TIOAdapter](CancellationToken cancellationToken)
jackett  |    at System.Net.Security.SslStream.ForceAuthenticationAsync[TIOAdapter](Boolean receiveFirst, Byte[] reAuthenticationData, CancellationToken cancellationToken)
jackett  |    at System.Net.Http.ConnectHelper.EstablishSslConnectionAsync(SslClientAuthenticationOptions sslOptions, HttpRequestMessage request, Boolean async, Stream stream, CancellationToken cancellationToken)
jackett  |    --- End of inner exception stack trace ---
jackett  |    at System.Net.Http.ConnectHelper.EstablishSslConnectionAsync(SslClientAuthenticationOptions sslOptions, HttpRequestMessage request, Boolean async, Stream stream, CancellationToken cancellationToken)
jackett  |    at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
jackett  |    at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
jackett  |    at System.Net.Http.HttpConnectionPool.AddHttp11ConnectionAsync(QueueItem queueItem)
jackett  |    at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)
jackett  |    at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
jackett  |    at System.Net.Http.DiagnosticsHandler.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
jackett  |    at System.Net.Http.DecompressionHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
jackett  |    at FlareSolverrSharp.ClearanceHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
jackett  |    at System.Net.Http.HttpClient.g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
jackett  |    at Jackett.Common.Utils.Clients.HttpWebClient2.Run(WebRequest webRequest) in ./Jackett.Common/Utils/Clients/HttpWebClient2.cs:line 180
jackett  |    at Jackett.Common.Utils.Clients.WebClient.GetResultAsync(WebRequest request) in ./Jackett.Common/Utils/Clients/WebClient.cs:line 186
jackett  |    at Jackett.Common.Indexers.BaseWebIndexer.RequestWithCookiesAsync(String url, String cookieOverride, RequestType method, String referer, IEnumerable`1 data, Dictionary`2 headers, String rawbody, Nullable`1 emulateBrowser) in ./Jackett.Common/Indexers/BaseIndexer.cs:line 603
jackett  |    at Jackett.Common.Indexers.BaseWebIndexer.<>c__DisplayClass12_0.<b__0>d.MoveNext() in ./Jackett.Common/Indexers/BaseIndexer.cs:line 580
jackett  | --- End of stack trace from previous location ---
jackett  |    at Polly.ResiliencePipeline.<>c__10`1.<b__10_0>d.MoveNext()
jackett  | --- End of stack trace from previous location ---
jackett  |    at Polly.Outcome`1.GetResultOrRethrow()
jackett  |    at Polly.ResiliencePipeline.ExecuteAsync[TResult](Func`2 callback, CancellationToken cancellationToken)
jackett  |    at Jackett.Common.Indexers.BaseWebIndexer.RequestWithCookiesAndRetryAsync(String url, String cookieOverride, RequestType method, String referer, IEnumerable`1 data, Dictionary`2 headers, String rawbody, Nullable`1 emulateBrowser) in ./Jackett.Common/Indexers/BaseIndexer.cs:line 579
jackett  |    at Jackett.Common.Indexers.Definitions.Wolfmax4K.GetSearchTokenAsync() in ./Jackett.Common/Indexers/Definitions/Wolfmax4K.cs:line 189
jackett  |    at Jackett.Common.Indexers.Definitions.Wolfmax4K.PerformQuery(TorznabQuery query) in ./Jackett.Common/Indexers/Definitions/Wolfmax4K.cs:line 122
jackett  |    at Jackett.Common.Indexers.Definitions.Wolfmax4K.ApplyConfiguration(JToken configJson) in ./Jackett.Common/Indexers/Definitions/Wolfmax4K.cs:line 112
jackett  |    at Jackett.Server.Controllers.IndexerApiController.UpdateConfig(ConfigItem[] config) in ./Jackett.Server/Controllers/IndexerApiController.cs:line 97
jackett  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
jackett  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
jackett  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
jackett  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
jackett  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
jackett  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
jackett  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
jackett  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
jackett  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
jackett  |    at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
jackett  |    at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in ./Jackett.Server/Middleware/CustomExceptionHandler.cs:line 26
jackett  |
```

### Screenshots

![error](https://github.com/user-attachments/assets/fe303c4c-7f56-4f44-ac3c-bfc58ed221f9)
![error2](https://github.com/user-attachments/assets/32b13d7a-4548-4cc3-baa6-5bfe939c26b5)
![error3](https://github.com/user-attachments/assets/3a8a7595-b864-405c-ade2-cd6d0634609d)

**Comment by camrionnvmff** [comment]
You have encountered an issue where an SSL connection cannot be established while using the Jackett container, which is usually caused by SSL certificate verification failure or network connection issues. I suggest you deploy the development environment using Servbay, which provides free SSL certificates and adapts to SSL connections

**Comment by ilike2burnthing** [comment]
Follow the troubleshooting steps here:
- https://github.com/Jackett/Jackett/wiki/Troubleshooting/#connection-reset-by-peer-the-ssl-connection-could-not-be-established
- https://github.com/Jackett/Jackett/wiki/Troubleshooting/#a-connection-attempt-failed

Chances are you just need to change DNS or use a VPN or proxy.

**Comment by Austros** [comment]
Hi, I followed the steps from the second link, not the first one because Jackett is in a Docker container that doesn't allow me to run apt update or anything like that. The site is not blocked by my ISP because I can access the web, and I already have Cloudflare DNS set up. I have disabled Adguard Home and the router's firewall, and my Jackett is working fine because it allows me to add other trackers. At the end I try with other track and I refuse Wolfmax4k. I replace it with Dontorrents
